src/pyscipopt/expr.pxi

```diff
@@ -1,3 +1,47 @@
+# In this file we implemenet the handling of expressions
+# We have two types of expressions: Expr and GenExpr.
+# The Expr can only handle polynomial expressions.
+# In addition, one can recover easily information from them.
+# A polynomial is a dictionary between `terms` and coefficients.
+# A `term` is a tuple of variables
+# For examples, 2*x*x*y*z - 1.3 x*y*y + 1 is stored as a
+# {Term(x,x,y,z) : 2, Term(x,y,y) : -1.3, Term() : 1}
+# Addition of common terms and expansion of exponents occur automatically.
+# Given the way `Expr`s are stored, it is easy to access the terms: e.g.
+# expr = 2*x*x*y*z - 1.3 x*y*y + 1
+# expr[Term(x,x,y,z)] returns 1.3
+# expr[Term(x)] returns 0.0
+#
+# On the other hand, when dealing with expressions more general than polynomials,
+# that is, absolute values, exp, log, sqrt or any general exponent, we use GenExpr.
+# GenExpr stores expression trees in a rudimentary way.
+# Basically, it stores the operator and the list of children.
+# We have different types of general expressions that in addition
+# to the operation and list of children stores
+# SumExpr: coefficients and constant
+# ProdExpr: constant
+# Constant: constant
+# VarExpr: variable
+# PowExpr: exponent
+# UnaryExpr: nothing
+# We do not provide any way of accessing the internal information of the expression tree,
+# nor we simplify common terms or do any other type of simplification.
+# The `GenExpr` is pass as is to SCIP and SCIP will do what it see fits during presolving.
+#
+# TODO: All this is very complicated, so we might wanna unify Expr and GenExpr.
```

Comment:
It seems to me like `GenExpr` is strictly more powerful than `Expr`, so `Expr` could be removed completely?
I don't know whether there is a difference in performance between these implementations...

Reply:
In any case, this step could also be done in the future.

Reply:
Well, this is not true, you can't ask information about GenExpr as you can with Expr, but if we do not care about that, then yes, it is more powerful.

I guess there are performance difference, I didn't really test this, but I would definitely do this in the future